slsDetectorSoftware/src/Detector.cpp

```diff
@@ -2067,6 +2067,69 @@ void Detector::setLEDEnable(bool enable, Positions pos) {
     pimpl->Parallel(&Module::setLEDEnable, pos, enable);
 }
 
+void Detector::setDacNames(const std::vector<std::string> names) {
+    if (getDetectorType().squash() != defs::CHIPTESTBOARD)
+        throw RuntimeError("Named dacs only for CTB");
+    if (names.size() != 18)
+        throw RuntimeError("Need to set all 18 dacs when naming dacs");
+
+    std::ofstream ofs("/dev/shm/slsDetectorPackage_ctbdacnames");
+    if (!ofs)
+        throw RuntimeError("Failed to open dacnames file in shared memory");
+
+    std::string s = sls::ToString(names);
+    ofs.write(&s[0], s.size());
+}
+
+std::vector<std::string> Detector::getDacNames() const {
+
+    auto type = getDetectorType().squash();
+    if (type == defs::CHIPTESTBOARD) {
+        try {
+            std::ifstream ifs("/dev/shm/slsDetectorPackage_ctbdacnames");
```

Comment:
Might need to include the SLSDETNAME. Might be part of this in the sharedMemory class.

Reply:
Now naming the file the same way as for the other shared memory. Using  _ctbdacs to identify it.